sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -142,45 +145,17 @@ export class ApplicationStudentsController extends BaseController {
       firstCOE,
     );
   }
-
   /**
-   * Submit an existing student application changing the status
-   * to submitted and triggering the necessary processes.
-   * The system will ensure that an application will always be
-   * transitioning from draft to submitted status. The student
-   * application is not supposed to be created directly in the
-   * submitted status in any scenario.
-   * @param payload payload to create the draft application.
-   * @param applicationId application id to be changed to submitted.
-   * @param studentToken token from the authenticated student.
+   * Validate the values in the submitted application before submitting.
+   * @param payload payload to create the application.
+   * @param programYear program year of the submitted application.
    */
-  @CheckSinValidation()
-  @Patch(":applicationId/submit")
-  @ApiUnprocessableEntityResponse({
-    description:
-      "Program Year is not active or " +
-      "Selected offering id is invalid or " +
-      "invalid study dates or selected study start date is not within the program year " +
-      "or APPLICATION_NOT_VALID or INVALID_OPERATION_IN_THE_CURRENT_STATUS or ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE " +
-      "or INSTITUTION_LOCATION_NOT_VALID or OFFERING_NOT_VALID " +
-      "or Invalid offering intensity",
-  })
-  @ApiBadRequestResponse({
-    description: "Form validation failed or Offering intensity type is invalid",
-  })
-  @ApiNotFoundResponse({ description: "Application not found." })
-  @ApiForbiddenResponse({
-    description: "You have a restriction on your account.",
-  })
-  async submitApplication(
+  private async validateSubmitApplicationData(
     @Body() payload: SaveApplicationAPIInDTO,
-    @Param("applicationId", ParseIntPipe) applicationId: number,
-    @UserToken() studentToken: StudentUserToken,
+    @Param() programYear: ProgramYear,
```

Comment:
The decorators body and param are not required for this method.

Reply:
As it validates the dynamic data, program year must be retrieved here and not sent as a param to the method. 

![image](https://github.com/bcgov/SIMS/assets/54600590/218bc870-448d-419a-a1e4-2c5bc39c64d9)